module-2/backend/week_two.md

```diff
@@ -26,20 +30,59 @@ end
 
 Now how would you find the owner of the team with an id of 4?
 
+> `Team.first.owner`
+
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+>Teacher `has_many` students and Student `has_and_belongs_to_many` teachers or just `belongs_to` Teacher depending on the school district I guess.
+
+
 6. Define foreign key, primary key, and schema.
+
+> foreign key is how one table accesses the relationship to another table. The primary key is the unique identifier of that specific table. Schema is the diagram of how those table are connected, through the foerign keys/ primary keys.
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+
+>it's how the tables access the relationships, ie `has_many`/`belongs_to`
+
 8. What are the parts of an HTTP response?
 
+> The verbs. `GET POST PUT DELETE` The `GET` request responds to a few actions. It will take one from the landing page, the index page, and the show page. For `POST PUT & DELET` there are two steps. First, the `GET` request for the specific `id` and then the corresponding verb for whatever the client has requested. `POST` for new content, `PUT` for updating content and `DELETE` for destroying content
+
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+>`pluck` grabs the attributes you want without loading whole records
+>`sum` adds the number of records for that specific column.
+>`minimum` returns the smallest occurence-returns the datatype of the column
+>`maximum` returns the largest occurence-returns the datatype of the column
+>`average` calculates averge value of a given column-returns the datatype of the column
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+
 3. What two columns does `t.timestamps null: false` create in our database?
+>`created_at` and `updated_at`
+
 4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+
+> School `has_many` teachers and a teacher `belongs_to` a school.
+
 5. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+
+>`school_id` as a foreign key on the teacher table.
+
 6. Give an example of when you might want to store information besides ids on a join table.
+
+>When that table itself has information that it needs to hold. In the students_teachers example, that joins table could have
+information about what class the teacher is teaching, the max class size allowed, the classroom that it takes place in, it's uartlerly budget...
+
 7. Describe and diagram the relationship between patients and doctors.
+
+>Doctors `has_many` patients and Patients `has_and_belongs_to_many` doctors
```

Comment:
For Patients: I would say pick one. Either they only have one doctor or they have many doctors. The `has_and_belongs_to_many` seems to be about creating a join without creating a model for the join table. I generally steer clear.